cgroup2/utils.go

```diff
@@ -378,55 +390,92 @@ func systemdUnitFromPath(path string) string {
 
 func readHugeTlbStats(path string) []*stats.HugeTlbStat {
 	usage := []*stats.HugeTlbStat{}
-	keyUsage := make(map[string]*stats.HugeTlbStat)
-	f, err := os.Open(path)
-	if err != nil {
-		return usage
-	}
-	files, err := f.Readdir(-1)
-	f.Close()
-	if err != nil {
-		return usage
+	for _, pagesize := range hugePageSizes() {
+		usage = append(usage, &stats.HugeTlbStat{
+			Max:      getStatFileContentUint64(filepath.Join(path, "hugetlb."+pagesize+".max")),
+			Current:  getStatFileContentUint64(filepath.Join(path, "hugetlb."+pagesize+".current")),
+			Pagesize: pagesize,
+		})
```

Comment:
Hmm! Can we do soemthing like
```
hugePageSizeSlice := hugePageSizes()
var usage = make(*stats.HugeTlbStat{}, len(hugePageSizeSlice)
and then iterate like an array
for idx, pageSize  := range hugePageSizeSlice {
  usage[idx] = .... 
}
```

Reply:
This is the same outcome though right? We still have a slice with empty entries, where the number of them is tied to how many huge page sizes we found in `/sys/kernel/mm/hugepages`. Before this we'd return an empty slice which I think is ideal

Reply:
Oh okay! yes. what I suggested matches your current code; but I was looking from an allocation perspective. If you pre-allocate; the slice won't have to expand while looping and appending.

I do not have an opinion on functionality as I do not know how users use it. I would try and keep the previous behavior though.

Reply:
I agree with @manugupt1 that pre-allocating the `usage` slice is better than growing it dynamically. Even if it won't shave off any allocations, this is more optimized overall, and since we know the size, it makes total sense to do so.

Reply:
Oh I agree about pre-allocating, more the point I was trying to get across still stands: this behavior is different than whats here today. We'll have entries in hugetlb stats regardless of if someone even has the controller enabled, as long as there's entries in /sys/kernel/mm/hugepages

Reply:
Right! I am not the best judge of this tbh!

Reply:
Lets see what others think, I don't think this is too crazy a departure